fall-2020/navigation.py

```diff
@@ -70,24 +66,117 @@ def set_desired_position(self, desired_position):
+        #checking if current postion and desired position are same and if same then prompting to input new values and calling navigate again
+        #if both positions are same then we could either take new input and run navigate again like shown below or we could return an error status like the boolean values for actuatuion requests as mentioned above.
+        if self.current_position == self.desired_position :
+            print("TBM is already at desired position or desired position not set,please set new desired position \n")
+            coordinate = input ("please input desired coordinate in the following form: ' x , y , z ' \n")
+            #coordinate value taken at one go and then seperated as inputting the coordinates seperately will take longer time
+            coordinate.replace(" ","")
+            desired_coordinate_list = coordinate.split(",")
+            desired_position_tuple = ()
+            for number_of_coordinates in range(3):
+                updater_tuple = (desired_coordinate_list[number_of_coordinates],)
+                desired_position_tuple = desired_position_tuple + updater_tuple
+            self.set_desired_position(desired_position_tuple)
+            self.navigate
+            return
+        #error checking for if the x given coordinate  is already behind the machine or not
+        if self.current_position(0) > self.desired_position(0): 
+            print("desired position for x axis is behind the TBM ,please put x coordinate which is infront of TBM \n")
+            coordinate = input ("please input desired coordinate in the following form: ' x , y , z ' \n")
+            coordinate.replace(" ","")
+            desired_coordinate_list = coordinate.split(",")
+            desired_position_tuple = ()
+            for number_of_coordinates in range(3):
+                updater_tuple = (desired_coordinate_list[number_of_coordinates],)
+                desired_position_tuple = desired_position_tuple + updater_tuple
+            self.set_desired_position(desired_position_tuple)
+            self.navigate
+            return
+            
+        targetreached = False
+        while targetreached != True :
+            #should add a pausing mechanism to let actuators take input and implement movement before checking new position and distance left to move using time.sleep(.) 
+            #confused about whether the steering methods return true if full movement to desired location complete or only if movement started and movement distance is given
+            #if steering methods return true after their full movement completed ,then loop is not required .
+            self.update_current_position()
+            #differences between the coordinates is calculated
+            difference_x = self.desired_position(0) - self.current_position(0)
+            difference_y = self.desired_position(1) - self.current_position(1)
+            difference_z = self.desired_position(2) - self.current_position(2)
```

Comment:
[] instead of () for accessing an element of the tuple

Reply:
I fixed it, thanks for pointing it out.